helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/AssignableNode.java

```diff
@@ -91,10 +91,21 @@ private void refresh(ClusterConfig clusterConfig, InstanceConfig instanceConfig,
       Collection<AssignableReplica> existingAssignment) {
     reset();
 
+    List<String> requiredCapacityKeys = clusterConfig.getInstanceCapacityKeys();
+
     Map<String, Integer> instanceCapacity = instanceConfig.getInstanceCapacityMap();
     if (instanceCapacity.isEmpty()) {
       instanceCapacity = clusterConfig.getDefaultInstanceCapacityMap();
     }
+    // Remove all the non-required capacity items from the map.
```

Comment:
Nit: Could we move this check to a private method? I think this line of logic separates nicely.

Reply:
I suggest creating a centralized capacity class that takes care of everything, (validation/init/updates), e.g, https://github.com/apache/helix/pull/452/files#diff-f45e90891858518de7280701a94da157

Reply:
@i3wangyi That is a good direction. But I believe we need some time to refine it. Let's use this implementation for now. In the near future, we are not planning to make capacity more complicated.
@narendly It is clear, but too many private methods are also confusing. I prefer to keep it in this way.

Reply:
It's already quite complicated to review. Make it a centralized IMO would only make things easier and more scalable to add/remove capacity keys.

Reply:
Sure, I don't have a strong preference here.